### PR TITLE
Support `TensorFrame` `concat` for complex stypes + test clean-up

### PR DESCRIPTION
**Pull request by weihua916**
Also, add more thorough testing of all complex stypes that we've added recently.

### CODE REVIEW

File: torch_frame/data/tensor_frame.py [hunk 1]
```diff
@@ -168,26 +169,28 @@ def __eq__(self, other: Any) -> bool:
...
-                    if isinstance(self_feat_value,
-                                  Tensor) and not torch.allclose(
-                                      self_feat_value, other_feat_value):
+            elif isinstance(self_feat, MultiNestedTensor):
+                if not isinstance(other_feat, MultiNestedTensor):
+                    return False
+                if not MultiNestedTensor.allclose(self_feat, other_feat):
+                    return False
+            elif isinstance(self_feat, Dict):
+                if not isinstance(other_feat, Dict):
```
Comment (akihironitta) [inline torch_frame/data/tensor_frame.py:187]: QQ: Any idea what the difference from built-in `dict` is when evaluating object with `isinstance`?
```suggestion
            elif isinstance(self_feat, dict):
                if not isinstance(other_feat, dict):
```

**Review by akihironitta — APPROVED** [review]
Neat!

File: torch_frame/data/tensor_frame.py (hunk 1 above)
Comment (weihua916) [reply torch_frame/data/tensor_frame.py:187]: I think they have the same behavior.
```python
>>> from typing import Dict
>>> isinstance({}, Dict)
True
>>> isinstance({}, dict)
True
>>>
```

Comment (akihironitta) [reply torch_frame/data/tensor_frame.py:187]: (cc @yiweny as I remember you also left a similar comment on my previous PR: `isinstance(obj, Tuple)` vs. `isinstance(obj, tuple)`.)

**Pull request merged**